api/app/controllers/spree/api/v2/storefront/products_controller.rb

```diff
@@ -43,11 +47,38 @@ def dependencies
             {
               collection_sorter:     Spree::Products::Sort,
               collection_finder:     Spree::Products::Find,
+              collection_paginator:  Spree::Products::Paginate,
               collection_serializer: Spree::V2::Storefront::ProductSerializer,
               resource_serializer:   Spree::V2::Storefront::ProductSerializer
             }
           end
 
+          def collection_options(collection)
+            {
+              links:   collection_links(collection),
+              meta:    collection_meta(collection),
+              include: collection_includes
+            }
+          end
+
+          def collection_links(collection)
+            {
+              self:  collection.current_page,
+              next:  collection.next_page || collection.total_pages,
```

Comment:
These methods are duplicated across this & taxons PR so let's centralize it somewhere to avoid duplication.